dealership/contracts.py

```diff
@@ -1,12 +1,44 @@
+interest_rate = {'car': 1.07, 'motorcycle': 1.03, 'truck': 1.11}
+leaseMultiplier = {'car': 1.2, 'motorcycle': 1, 'truck': 1.7}
+
 class Contract(object):
-    pass
+  def __init__(self, vehicle, customer):
+    self.vehicle = vehicle 
+    self.customer = customer
+    self.total = 0
+  
+  def total_value(self):
+    sales_price = self.vehicle.sale_price()
+    if self.contract == 'buy': 
+      I = interest_rate[self.vehicle.name]
+      self.total = sales_price + ((I * self.monthly_payments * sales_price) / 100)
+    else: 
+      lease_multiplier = (sales_price * leaseMultiplier[self.vehicle.name] / self.length_in_months)
+      self.total = sales_price + lease_multiplier
+    if self.customer.is_employee(): 
+       self.total=  self.total * 0.9
+    return self.total
```

Comment:
so `Contract` is your parent.. and it should not need to 'know' the existence of `BuyContract` or `LeaseContract`... `Contract` should have a generic `total_value(self)` which has as much code as all child classes can reuse...